source/developer/developer-setup.md

```diff
@@ -174,34 +172,56 @@ Any issues? Please let us know on our forums at: http://forum.mattermost.org
     - `env GOOS=darwin GOARCH=amd64 go install std`
     - `env GOOS=linux GOARCH=amd64 go install std`
 
 Any issues? Please let us know on our forums at: http://forum.mattermost.org
 
-## Developing Using PostgreSQL
+## Troubleshooting
+
+#### I get the following error when running `make run` on Mac OS X: "Cannot connect ot the Docker daemon"
 
-By default, development is done using a MySQL database. To switch your developement instance to use a PostgreSQL database, update the following settings in the `SqlSettings` section of your `config/config.json`
+If you have Docker Tools installed (as opposed to Docker for Mac), you need make sure `docker-machine` is running with the following:
 ```
-    "DriverName": "mysql",
-    "DataSource": "mmuser:mostest@tcp(dockerhost:3306)/mattermost_test?charset=utf8mb4,utf8",
+docker-machine start dev
 ```
-to
+
+#### I get the following error when running `make run`: "Failed to ping db err:dial tcp 192.168.99.100:3306: getsockopt: connection refused"
+
+It appears that your MySQL database isn't running. If you run `docker ps`, you should see a line like
 ```
-    "DriverName": "postgres",
-    "DataSource": "postgres://mmuser:mostest@dockerhost:5432?sslmode=disable&connect_timeout=10",
+ecb17c10973d    mysql:5.7  "/entrypoint.sh mysql"   2 weeks ago    Up 24 hours  0.0.0.0:3306->3306/tcp     mattermost-mysql
+```
+If not, running `make clean-docker` will remove all existing docker containers so that they'll be recreated next time you call `make run`.
+
+#### I get the following error when running `make run`: "Error starting server, err:listen tcp :8065: bind: address already in use"
+
+There's likely another Mattermost instance already running. You can use `make stop` to stop it before running `make run` again.
+
+If there isn't another copy of Mattermost running and you need to change the port that Mattermost is running on, you can do so by changing the `ListenAddress` setting in the `ServiceSettings` section of `config/config.json`.
+
+#### I don't see any error messages, but I can't access `http://localhost:8065`
+
+It's possible that the server reported an error, but it was missed because of all of the output from the Javascript compiler. Try running `make run-server` by itself to see its output. If you still don't see any error messages, continue to the next section.
+
+#### I don't see anything logged to the console when Mattermost is running
+
+You can enable console logging in the `LogSettings` section of your `config/config.json` by setting `EnableConsole` to `true`.
+
+#### I can't log into Mattermost because I don't have an account
+
+You can create an account using the following command:
+```
+go run mattermost.go -create_user -email="user@example.com" -password="mypassword"
+```
+Optionally, you can make that account a System Admin with the following command:
+```
+go run mattermost.go -assign_role -email="user@example.com" -role="system_admin"
```

Comment:
I think you need to have the role as `system_admin system_user` since assigning roles will overwrite all the user's previous roles